propolis/src/hw/pci/bus.rs

```diff
@@ -204,26 +209,112 @@ impl Inner {
             }
         }
     }
+
+    fn extended_config_rw(&self, _addr: usize, rwo: RWOp, ctx: &DispCtx) {
+        use crate::{
+            common::{ReadOp, WriteOp},
+            hw::pci::bits::{LEN_CFG, MASK_ECAM_ACCESS_ALIGN},
+        };
```

Comment:
Perhaps make something akin to the `PciCfgDecoder`, where we can express things like the max bus for ECAM, which would in turn determine its sizing and alignment requirements for the MMIO space registration and handling.

Reply:
Done. Thanks especially for this suggestion--I think it substantially simplifies things.